output/terminal_ncurses.c

```diff
@@ -1,145 +1,142 @@
-
-	getmaxyx(stdscr,*h,*w);
-	clear(); //clearing in case of resieze
+void get_terminal_dim_ncurses(int* width, int* height) {
+	getmaxyx(stdscr, *height, *width);
+	clear(); // clearing in case of resieze
 }
 
+#define TERMINAL_RESIZED -1
 
-int draw_terminal_ncurses(int tty, int h, int w, int bars, int bw, int bs, int rest, int f[200], int flastd[200]) {
-
-	int i, n, q;
-	const wchar_t* bh[] = {L"\u2581", L"\u2582", L"\u2583", L"\u2584", L"\u2585", L"\u2586", L"\u2587", L"\u2588"};
+int draw_terminal_ncurses(int is_tty, int terminal_height, int terminal_width,
+int bars_count, int bar_width, int bar_spacing, int rest, const int f[200],
+int flastd[200]) {
+	const wchar_t* bar_heights[] = {L"\u2581", L"\u2582", L"\u2583",
+		L"\u2584", L"\u2585", L"\u2586", L"\u2587", L"\u2588"};
+	#define LAST ((sizeof(bar_heights) / sizeof(bar_heights[0])) - 1)
 
 	// output: check if terminal has been resized
-	if (!tty) {
-		if ( LINES != h || COLS != w) {
-			return -1;
+	if (!is_tty) {
+		if (LINES != terminal_height || COLS != terminal_width) {
+			return TERMINAL_RESIZED;
 		}
 	}
-
-	h = h - 1;
-
-	for (i = 0; i <  bars; i++) {
-
-		if(f[i] > flastd[i]){//higher then last one
-			if (tty) for (n = flastd[i] / 8; n < f[i] / 8; n++) for (q = 0; q < bw; q++) mvprintw((h - n), i*bw + q + i*bs + rest, "%d",8);
-			else for (n = flastd[i] / 8; n < f[i] / 8; n++) for (q = 0; q < bw; q++) mvaddwstr((h - n), i*bw + q + i*bs + rest, bh[7]);
-			if (f[i] % 8 != 0) {
-				if (tty) for (q = 0; q < bw; q++) mvprintw( (h - n), i*bw + q + i*bs + rest, "%d",(f[i] % 8) );
-				else for (q = 0; q < bw; q++) mvaddwstr( (h - n), i*bw + q + i*bs + rest, bh[(f[i] % 8) - 1]);
+	int height = terminal_height - 1;
+	int n;
+	#define CURRENT_COLUMN bar*bar_width + width + bar*bar_spacing + rest
+	for (int bar = 0; bar < bars_count; bar++) {
+		if (f[bar] > flastd[bar]) { // higher then last one
+			if (is_tty) {
+				for (n = flastd[bar] / 8; n < f[bar] / 8; n++)
+					for (int width = 0; width < bar_width; width++)
+						mvprintw((height - n), CURRENT_COLUMN, "%d", 8);
+			} else {
+				for (n = flastd[bar] / 8; n < f[bar] / 8; n++)
+					for (int width = 0; width < bar_width; width++)
+						mvaddwstr((height - n), CURRENT_COLUMN,
+								bar_heights[LAST]);
 			}
-		}else if(f[i] < flastd[i]){//lower then last one
-			for (n = f[i] / 8; n < flastd[i]/8 + 1; n++) for (q = 0; q < bw; q++) mvaddstr( (h - n), i*bw + q + i*bs + rest, " ");
-			if (f[i] % 8 != 0) {
-				if (tty) for (q = 0; q < bw; q++) mvprintw((h - f[i] / 8), i*bw + q + i*bs + rest, "%d",(f[i] % 8) );
-				else for (q = 0; q < bw; q++) mvaddwstr((h - f[i] / 8), i*bw + q + i*bs + rest, bh[(f[i] % 8) - 1]);
+			if (f[bar] % 8) {
+				if (is_tty) {
+					for (int width = 0; width < bar_width; width++)
+						mvprintw((height - n), CURRENT_COLUMN, "%d",
+								(f[bar] % 8));
+				} else {
+					for (int width = 0; width < bar_width; width++)
+						mvaddwstr((height - n), CURRENT_COLUMN,
```

Comment:
Here is 'n' which is not inside its loop. It supposed to be 'f[bar] / 8' or what?